docssrc/src/commandarguments.md

```diff
@@ -0,0 +1,202 @@
+# CommandArguments
+
+The `CommandArguments` class was introduced in CommandAPI 9.0.0 and provides a much more powerful way of accessing arguments than just an array of arguments which existed until 9.0.0.
+
+While the argument array just gives the possibility to access the arguments via the array notation (`args[0]`), the `CommandArguments` class offers much more, including:
+
+- [Access the inner structure directly](#access-the-inner-structure-directly)
+- [What terms are used?](#what-terms-are-used)
+  - [`nodeName`](#nodename)
+  - [`raw argument`](#raw-argument)
+  - [`unsafe argument`](#unsafe-argument)
+- [Access arguments](#access-arguments)
+  - [Access arguments by node name](#access-arguments-by-node-name)
+  - [Access arguments by index](#access-arguments-by-index)
+- [Access raw arguments](#access-raw-arguments)
+  - [Access raw arguments by node name](#access-raw-arguments-by-node-name)
+  - [Access raw arguments by index](#access-raw-arguments-by-index)
+- [Access unsafe arguments](#access-unsafe-arguments)
+  - [Access arguments by node name](#access-arguments-by-node-name-1)
+  - [Access arguments by index](#access-arguments-by-index-1)
+
+ATTENTION!!!!! Every executable and generated documentation has to be deleted before committing anything!
+
+-----
+
+## Access the inner structure directly
+
+To provide arguments, the `CommandArguments` class stores:
+
+- a `Object[]` of parsed arguments
+- a `Map<String, Object>` of parsed arguments mapped to their node names
+- a `String[]` of raw arguments
+- a `Map<String, String>` of raw arguments mapped to their node names
+- a `String` which holds the full input
+
+Although not recommended, it is possible to access these fields directly with methods the `CommandArguments` class provides:
+
+```java
+Object[] args();                  // Returns the argument array
+Map<String, Object> argsMap();    // Returns an unmodifiable map containing the arguments mapped to their node names
+String[] rawArgs();               // Returns the raw argument array
+Map<String, String> rawArgsMap(); // Returns an unmodifiable map containing the raw arguments mapped to their node names
+String getFullInput();               // Returns the full command string (including the /)
+```
+
+Additionally, the `CommandArguments` class has one more method that isn't directly backed by a field and returns the amount of arguments for a command:
+
+```java
+int count();
+```
+
+While these methods can be used to access arguments, it may be safer to use the other methods the `CommandArguments` class provides to access arguments.
+
+-----
+
+## What terms are used?
+
+Throughout this page, multiple terms are used that may need an explanation.
+
+### `nodeName`
+
+The `nodeName` is set when initializing an argument. For example:
+
+```java
+new StringArgument("string") 
+```
+
+The `nodeName` here would be `string`.
```

Comment:
This should probably not appear in this page. I think it may be better to appear in `arguments.md` where we mention "node" for the first time.